woocommerce/woocommerce-android
# Fix analytics cache

Closes: #11957

### Description
This PR refactors the local analytic cache to align with the changes introduced in the Analytic Cards configuration.

Context:
In the previous version, the Analytic Hub screen relied on the last update value for Analytic Data: `All` to determine whether to display cached data or fetch new data. With the new feature allowing users to configure visible cards on the analytic screen, the concept of "All" became obsolete, as not all card data is necessarily needed (some cards may be hidden).

Changes Introduced:

- Remove "All" Value:
  - Eliminated the "ALL" value from the cache logic.
- Update shouldUpdateFunction
  - Modified the shouldUpdateFunction to evaluate the date of each analytic data individually.
  - The function now checks if the last update value for each visible card is recent enough to use cached data.
  - If the cached data is outdated, the function fetches new data from the API.

### Testing information

1. Open the app
2. Tap on one of the dashboard cards to navigate to the analytic hub screen
3. Check that after initially loading a time selection, the second time the time selection is loaded, data is presented from the local cache. 

### Images/gif

https://github.com/woocommerce/woocommerce-android/assets/18119390/19bb3d12-9467-4726-99c4-d6c67e08a7d1
